go/libkb/merkle_client.go

```diff
@@ -772,33 +838,55 @@ func (mc *MerkleClient) lookupPathAndSkipSequenceHelper(m MetaContext, q HTTPArg
 		q.Add("sig_hints_low", I{sigHints.version})
 	}
 
-	// Get back a series of skips from the last merkle root we had to the new
-	// one we're getting back, and hold the server to it.
-	lastSeqno := lastRoot.Seqno()
-	if lastSeqno != nil {
-		q.Add("last", I{int(*lastSeqno)})
+	// Get back a path from the leaf to the current merkle root. The root itself
+	// is not included.
+	tail := root.Seqno()
+	if tail != nil {
+		q.Add("tail", I{int(*tail)})
 	}
 
 	apiRes, err = m.G().API.Get(m, APIArg{
 		Endpoint:       "merkle/path",
 		SessionType:    APISessionTypeOPTIONAL,
 		Args:           q,
-		AppStatusCodes: []int{SCOk, SCNotFound, SCDeleted},
+		AppStatusCodes: []int{SCOk, SCNotFound, SCDeleted, SCMerkleUpdateRoot},
 	})
 
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	refreshRootAndRetry := func() (apiRes *APIRes, newRoot *MerkleRoot, err error) {
+		root, err = mc.FetchRootFromServer(m, 0)
+		if err != nil {
+			return nil, nil, err
+		}
+		return mc.lookupLeafAndPathHelper(m, q, sigHints, root, opts)
```

Comment:
could server cause a infinite loop here?

Reply:
Yes, but it can also not reply or take a long time or whatever. Do you think this is a risk?

Reply:
Was thinking more of a server bug than server being malicious

Reply:
My main concern with this is that someone has a script continuously creating paper keys or following people and you incidentally try to load them this might generate a lot of network traffic/computation for nothing. I would do the retry with polling disabled as you suggested. However @maxtaco argued that this is safer as you are really sure to get the very latest version of a user this way.